Sources/FluentUI_macOS/Core/Extensions/NSColor+Extensions.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+#if canImport(AppKit)
+import AppKit
+import SwiftUI
+
+extension NSColor {
+
+    /// Creates a dynamic color object that returns the appropriate color value based on the current
+    /// rendering context.
+    ///
+    /// The decision order for choosing between the colors is based on the following questions, in order:
+    /// - Is the current `userInterfaceStyle` `.dark` or `.light`?
+    ///
+    /// - Parameter light: The default color for a light context. Required.
+    /// - Parameter dark: The override color for a dark context. Optional.
+    @objc public convenience init(light: NSColor,
+                                  dark: NSColor? = nil) {
+        self.init(name: nil) { appearance -> NSColor in
```

Comment:
Could possibly use https://developer.apple.com/documentation/appkit/nscolor/init(name:dynamicprovider:) ?

Reply:
Yep, that's what this method is! Just using the implicit closure attachment thing in Swift.

Reply:
Dang. I can't read apparently 😂